src/zarr/testing/store.py

```diff
@@ -1,61 +1,121 @@
+from typing import Generic, TypeVar
+
 import pytest
 
 from zarr.abc.store import Store
 from zarr.buffer import Buffer
+from zarr.store.core import _normalize_interval_index
 from zarr.testing.utils import assert_bytes_equal
 
+S = TypeVar("S", bound=Store)
+
+
+class StoreTests(Generic[S]):
+    store_cls: type[S]
 
-class StoreTests:
-    store_cls: type[Store]
+    def set(self, store: S, key: str, value: Buffer) -> None:
+        """
+        Insert key: value pairs into a store without using the store methods.
+        """
+        raise NotImplementedError
+
+    def get(self, store: S, key: str) -> Buffer:
+        """
+        Retrieve values from a store without using the store methods.
+        """
+
+        raise NotImplementedError
```

Comment:
@jhamman I added these two methods so that `store.get` and `store.set` could be tested independently. i hope this functionality is consistent with your intention for this class.